src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -180,10 +181,22 @@ const convertFrameworkSummary = (framework) => {
 const convertTaskDetail = async (taskStatus, ports, logPathPrefix) => {
   // get container ports
   const containerPorts = {};
-  if (ports) {
+  if (ports && taskStatus.attemptStatus.podUID) {
     const randomPorts = JSON.parse(ports);
-    for (let port of Object.keys(randomPorts)) {
-      containerPorts[port] = randomPorts[port].start + taskStatus.index * randomPorts[port].count;
+    if (randomPorts.ports) {
+      for (let port of Object.keys(randomPorts.ports)) {
+        const portNums = [...Array(randomPorts.ports[port].count).keys()].map((index) => {
+          const rawString = taskStatus.attemptStatus.podUID + port + index;
+          return parseInt(crypto.createHash('md5').update(rawString)
+            .digest('hex').substring(0, 12), 16) % (randomPorts.portEnd - randomPorts.portStart) + randomPorts.portStart;
```

Comment:
Why not just parseInt(crypto.createHash('md5').update(rawString).digest('hex'), 64) ?
This will not lost more hash info.

Reply:
It's because for js the int range is `253 - 1`. Refer to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER.

For runtime part, the `number` don't have such limit. If don't truncate the string, it will cause different port number in rest-server and runtime.

Reply:
Fold the md5 to within range

Reply:
Fixed